twisted/web/test/test_http2.py

```diff
@@ -1403,6 +1403,32 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_resetAfterBody(self):
+        """
+        A client that immediately resets after sending the body causes Twisted
+        to send no response.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        requestBytes = f.preamble()
+        requestBytes += buildRequestBytes(self.getRequestHeaders, [], f)
+        requestBytes += hyperframe.frame.RstStreamFrame(
+            stream_id=1
+        ).serialize()
+        a.makeConnection(transport)
+        a.dataReceived(requestBytes)
+
+        buffer = FrameBuffer()
+        buffer.receiveData(transport.value())
+        frames = list(buffer)
+
+        self.assertEqual(len(frames), 1)
```

Comment:
Reading the RFC this looks like the `server connection preface`,

```
The server connection preface consists of a potentially empty
SETTINGS frame (Section 6.5) that MUST be the first frame the server
sends in the HTTP/2 connection.
```


So to have a DSL, maybe this can be implemented as `self.assertServerConnectionPreface(frames[1])`